utils/s2n_ensure.c

```diff
@@ -15,7 +15,7 @@
 
 #include "utils/s2n_safety.h"
 
-void *s2n_ensure_memcpy_trace(void *restrict to, const void *restrict from, size_t size)
+void *s2n_ensure_memmove_trace(void *to, const void *from, size_t size)
```

Comment:
Is there a reason why this function is called _trace?

Reply:
History: https://github.com/aws/s2n-tls/blame/9014236e7ba83869251298341264ccf1a37cd9f6/utils/s2n_ensure.c#L18-L24 Looks like it used to manually set the debug string.

But now the only work it really does is to check the result of memmove. The higher level macros could probably just do that themselves.

I opened an issue to remove it: https://github.com/aws/s2n-tls/issues/4457 I'd still like to merge this, as it's a step in the right direction.